src/measure_theory/decomposition/signed_hahn.lean

```diff
@@ -0,0 +1,627 @@
+
+lemma set.pairwise_disjoint_on_inter {f : ℕ → set α} {a : set α}
+  (hf : pairwise (disjoint on f)) : pairwise (disjoint on (λ n, a ∩ f n)) :=
+begin
+  rintro i j hij x ⟨⟨-, hx₁⟩, -, hx₂⟩,
+  exact hf i j hij ⟨hx₁, hx₂⟩,
+end
+
+lemma set.pairwise_disjoint_on_inter_disjointed {f : ℕ → set α} {a : set α} :
+  pairwise $ disjoint on (λ n, a ∩ disjointed f n) :=
+set.pairwise_disjoint_on_inter disjoint_disjointed
+
+lemma tendsto_top_of_pos_summable_inv {f : ℕ → ℝ}
+  (hf : summable f⁻¹) (hf' : ∀ n, 0 < f n) : tendsto f at_top at_top :=
+begin
+  rw [show  f = f⁻¹⁻¹, by { ext, simp }],
+  apply filter.tendsto.inv_tendsto_zero,
+  apply tendsto_nhds_within_of_tendsto_nhds_of_eventually_within _
+    (summable.tendsto_at_top_zero hf),
+  rw eventually_iff_exists_mem,
+  refine ⟨set.Ioi 0, Ioi_mem_at_top _, λ _ _, _⟩,
+  rw [set.mem_Ioi, inv_eq_one_div, one_div, pi.inv_apply, _root_.inv_pos],
+  exact hf' _,
+end
+
+lemma exists_tendsto_Inf {S : set ℝ} (hS : ∃ x, x ∈ S) (hS' : ∃ x, ∀ y ∈ S, x ≤ y) :
+  ∃ (f : ℕ → ℝ) (hf : ∀ n, f n ∈ S), tendsto f at_top (nhds (Inf S)) :=
+begin
+  have : ∀ n : ℕ, ∃ t ∈ S, t < Inf S + 1 / (n + 1 : ℝ),
+  { exact λ n, (real.Inf_lt _ hS hS').1 ((lt_add_iff_pos_right _).2 nat.one_div_pos_of_nat) },
+  choose f hf using this,
+  refine ⟨f, λ n, (hf n).1, _⟩,
+  rw tendsto_iff_dist_tendsto_zero,
+  refine squeeze_zero_norm _ tendsto_one_div_add_at_top_nhds_0_nat,
+  intro n,
+  obtain ⟨hf₁, hf₂⟩ := hf n,
+  rw [real.dist_eq, real.norm_eq_abs, abs_abs,
+      abs_of_nonneg (sub_nonneg.2 (real.Inf_le S hS' hf₁))],
+  linarith,
+end
+
+lemma exists_tendsto_Sup {S : set ℝ} (hS : ∃ x, x ∈ S) (hS' : ∃ x, ∀ y ∈ S, y ≤ x) :
+  ∃ (f : ℕ → ℝ) (hf : ∀ n, f n ∈ S), tendsto f at_top (nhds (Sup S)) :=
+begin
+  have : ∀ n : ℕ, ∃ t ∈ S, Sup S - 1 / (n + 1 : ℝ) < t,
+  { intro n,
+    apply (real.lt_Sup _ hS hS').1,
+    rw [sub_lt, _root_.sub_self],
+    exact nat.one_div_pos_of_nat },
+  choose f hf using this,
+  refine ⟨f, λ n, (hf n).1, _⟩,
+  rw tendsto_iff_dist_tendsto_zero,
+  refine squeeze_zero_norm _ tendsto_one_div_add_at_top_nhds_0_nat,
+  intro n,
+  obtain ⟨hf₁, hf₂⟩ := hf n,
+  rw [real.dist_eq, real.norm_eq_abs, abs_abs, abs_of_nonpos],
+  { linarith },
+  { rw [sub_le, _root_.sub_zero],
+    exact real.le_Sup S hS' hf₁ },
+end
+
+lemma nat.sub_one_lt {n : ℕ} (hn : 1 ≤ n) : n - 1 < n :=
+begin
+  induction n with k hk,
+  { norm_num at hn },
+  { rw [nat.succ_sub_succ_eq_sub, nat.sub_zero], exact lt_add_one k }
+end
+
+end lemmas
+
+variables {α β : Type*} [measurable_space α]
+variables {M : Type*} [add_comm_monoid M] [topological_space M] [ordered_add_comm_monoid M]
+
+namespace measure_theory
+
+namespace vector_measure
+
+/-- A set `i` is positive with respect to a vector measure if for all
+measurable set `j`, `j ⊆ i`, `j` has non-negative measure. -/
+def positive (v : vector_measure α M) (i : set α) : Prop :=
+∀ j ⊆ i, measurable_set j → 0 ≤ v j
```

Comment:
You might want to try changing the first binder to be implicit, which makes this:
```suggestion
∀ ⦃j⦄, j ⊆ i → measurable_set j → 0 ≤ v j
```
It would remove some underscores below, but you'd have to try it to see if it makes any of your other proofs harder.

Reply:
Thanks! I was able to remove a dozen or so underscores without anything breaking